scan/lib/scan/test_command_generator.rb

```diff
@@ -49,7 +49,9 @@ def actions
         actions = []
         actions << :clean if config[:clean]
         actions << :build unless config[:skip_build]
-        actions << :test
+        actions << "build-for-testing" if config[:build_for_testing]
+        actions << :test unless config[:test_without_building] || config[:build_for_testing]
```

Comment:
unless in combination with the `or` is pretty confusing, and caused a major issue in gym already. Let's rewrite that line so it's easier to read and more clear